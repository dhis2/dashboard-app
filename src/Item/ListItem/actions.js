import {
    acUpdateDashboardItem,
    acRemoveDashboardItem,
} from '../../actions/editDashboard';

export const tRemoveListItemContent = (dashboardItem, content) => dispatch => {
    const listItemType = dashboardItem.type.toLowerCase();

    const newContent = dashboardItem[listItemType].filter(
        item => item.id !== content.id
    );

    if (newContent.length) {
        dashboardItem[listItemType] = newContent;
        dispatch(acUpdateDashboardItem(dashboardItem));
    } else {
        dispatch(acRemoveDashboardItem(dashboardItem));
    }
};
