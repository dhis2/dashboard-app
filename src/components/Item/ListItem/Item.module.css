.list {
    margin-top: 10px;
}

.item {
    margin-bottom: 8px;
}

.itemicon {
    vertical-align: text-bottom;
    width: 18px;
    height: 18px;
}

.link {
    font-size: 14px;
    margin-left: 5px;
    text-decoration: underline;
}

.deleteicon {
    vertical-align: text-bottom;
}

.deletebutton {
    border: none;
    border-radius: 50%;
    cursor: pointer;
    margin-left: 5px;
    padding: 0px 5px;
    vertical-align: text-bottom;
}

.deletebutton:focus {
    outline: none;
}
