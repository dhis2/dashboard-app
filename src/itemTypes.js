import { getBaseUrl } from './util';

// Item types
export const REPORT_TABLE = 'REPORT_TABLE';
export const CHART = 'CHART';
export const MAP = 'MAP';
export const EVENT_REPORT = 'EVENT_REPORT';
export const EVENT_CHART = 'EVENT_CHART';
export const APP = 'APP';
export const REPORTS = 'REPORTS';
export const RESOURCES = 'RESOURCES';
export const USERS = 'USERS';
export const MESSAGES = 'MESSAGES';
export const TEXT = 'TEXT';
export const SPACER = 'SPACER';

// Domain types
export const DOMAIN_TYPE_AGGREGATE = 'AGGREGATE';
export const DOMAIN_TYPE_TRACKER = 'TRACKER';

// Visualization types
export const VISUALIZATION_TYPE_TABLE = 'TABLE';
export const VISUALIZATION_TYPE_CHART = 'CHART';
export const VISUALIZATION_TYPE_MAP = 'MAP';

export const spacerContent = 'SPACER_ITEM_FOR_DASHBOARD_LAYOUT_CONVENIENCE';
export const emptyTextItemContent = 'TEXT_ITEM_WITH_NO_CONTENT';
export const isSpacerType = item =>
    item.type === TEXT && item.text === spacerContent;
export const isTextType = item =>
    item.type === TEXT && item.text !== spacerContent;
export const isPluginType = item =>
    itemTypeMap[item.type].hasOwnProperty('plugin');

// Item type map
export const itemTypeMap = {
    [REPORT_TABLE]: {
        id: REPORT_TABLE,
        endPointName: 'reportTables',
        propName: 'reportTable',
        countName: 'reportTableCount',
        pluralTitle: 'Report tables',
        plugin: global.reportTablePlugin,
        domainType: DOMAIN_TYPE_AGGREGATE,
        visualizationType: VISUALIZATION_TYPE_TABLE,
        appUrl: id => `dhis-web-pivot/?id=${id}`,
        icon: 'ViewList',
    },
    [CHART]: {
        id: CHART,
        endPointName: 'charts',
        propName: 'chart',
        countName: 'chartCount',
        pluralTitle: 'Charts',
        plugin: global.chartPlugin,
        domainType: DOMAIN_TYPE_AGGREGATE,
        visualizationType: VISUALIZATION_TYPE_CHART,
        appUrl: id => `dhis-web-visualizer/?id=${id}`,
        icon: 'InsertChart',
    },
    [MAP]: {
        id: MAP,
        endPointName: 'maps',
        propName: 'map',
        countName: 'mapCount',
        pluralTitle: 'Maps',
        plugin: global.mapPlugin,
        domainType: DOMAIN_TYPE_AGGREGATE,
        visualizationType: VISUALIZATION_TYPE_MAP,
        appUrl: id => `dhis-web-maps/?id=${id}`,
        icon: 'Public',
    },
    [EVENT_REPORT]: {
        id: EVENT_REPORT,
        endPointName: 'eventReports',
        propName: 'eventReport',
        countName: 'eventReportCount',
        pluralTitle: 'Event reports',
        plugin: global.eventReportPlugin,
        domainType: DOMAIN_TYPE_TRACKER,
        visualizationType: VISUALIZATION_TYPE_TABLE,
        appUrl: id => `dhis-web-event-reports/?id=${id}`,
        icon: 'ViewList',
    },
    [EVENT_CHART]: {
        id: EVENT_CHART,
        endPointName: 'eventCharts',
        propName: 'eventChart',
        countName: 'eventChartCount',
        pluralTitle: 'Event charts',
        plugin: global.eventChartPlugin,
        domainType: DOMAIN_TYPE_TRACKER,
        visualizationType: VISUALIZATION_TYPE_CHART,
        appUrl: id => `dhis-web-event-visualizer/?id=${id}`,
        icon: 'InsertChart',
    },
    [APP]: {
        endPointName: 'apps',
        propName: 'app',
        countName: 'appCount',
        pluralTitle: 'Apps',
        icon: 'Extension',
    },
    [REPORTS]: {
        id: REPORTS,
        endPointName: 'reports',
        propName: 'reports',
        countName: 'reportCount',
        pluralTitle: 'Reports',
        appUrl: id =>
            `dhis-web-reporting/getReportParams.action?mode=report&uid=${id}`,
        icon: 'ViewList',
    },
    [RESOURCES]: {
        id: RESOURCES,
        endPointName: 'resources',
        propName: 'resources',
        countName: 'resourceCount',
        pluralTitle: 'Resources',
        appUrl: id => `api/documents/${id}`,
        icon: 'Description',
    },
    [USERS]: {
        id: USERS,
        endPointName: 'users',
        propName: 'users',
        countName: 'userCount',
        pluralTitle: 'Users',
        appUrl: id => `dhis-web-dashboard-integration/profile.action?id=${id}`,
        icon: 'Person',
    },
    [TEXT]: {
        id: TEXT,
        propName: 'text',
        icon: 'FontDownload',
    },
    [MESSAGES]: {
        propName: 'messages',
        icon: 'Email',
    },
    [SPACER]: {
        propName: 'text',
        icon: 'CropFree',
    },
};

export const getItemUrl = (type, id, d2) => {
    if (itemTypeMap[type] && itemTypeMap[type].appUrl) {
        return `${getBaseUrl(d2)}/${itemTypeMap[type].appUrl(id)}`;
    }
};
