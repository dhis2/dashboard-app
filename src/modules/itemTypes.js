import i18n from '@dhis2/d2-i18n';
import TableIcon from '@material-ui/icons/ViewList';
import ChartIcon from '@material-ui/icons/InsertChart';
import MapIcon from '@material-ui/icons/Public';
import ExtensionIcon from '@material-ui/icons/Extension';
import DescriptionIcon from '@material-ui/icons/Description';
import PersonIcon from '@material-ui/icons/Person';
import FontDownloadIcon from '@material-ui/icons/FontDownload';
import EmailIcon from '@material-ui/icons/Email';
import CropFreeIcon from '@material-ui/icons/CropFree';
import NotInterestedIcon from '@material-ui/icons/NotInterested';

import { getBaseUrl } from './util';

// Item types
export const REPORT_TABLE = 'REPORT_TABLE';
export const CHART = 'CHART';
export const MAP = 'MAP';
export const EVENT_REPORT = 'EVENT_REPORT';
export const EVENT_CHART = 'EVENT_CHART';
export const APP = 'APP';
export const REPORTS = 'REPORTS';
export const RESOURCES = 'RESOURCES';
export const USERS = 'USERS';
export const MESSAGES = 'MESSAGES';
export const TEXT = 'TEXT';
export const SPACER = 'SPACER';

// Domain types
export const DOMAIN_TYPE_AGGREGATE = 'AGGREGATE';
export const DOMAIN_TYPE_TRACKER = 'TRACKER';

// Visualization types
export const VISUALIZATION_TYPE_TABLE = 'TABLE';
export const VISUALIZATION_TYPE_CHART = 'CHART';
export const VISUALIZATION_TYPE_MAP = 'MAP';

export const spacerContent = 'SPACER_ITEM_FOR_DASHBOARD_LAYOUT_CONVENIENCE';
export const emptyTextItemContent = 'TEXT_ITEM_WITH_NO_CONTENT';
export const isSpacerType = item =>
    item.type === TEXT && item.text === spacerContent;
export const isTextType = item =>
    item.type === TEXT && item.text !== spacerContent;
export const isPluginType = item =>
    itemTypeMap[item.type].hasOwnProperty('plugin');

export const getPlugin = type => {
    const pluginName = itemTypeMap[type].plugin;

    if (type === CHART) {
        return true;
    }

    return global[pluginName];
};

// Item type map
export const itemTypeMap = {
    [REPORT_TABLE]: {
        id: REPORT_TABLE,
        endPointName: 'reportTables',
        propName: 'reportTable',
        pluralTitle: i18n.t('Pivot tables'),
        plugin: 'reportTablePlugin',
        domainType: DOMAIN_TYPE_AGGREGATE,
        visualizationType: VISUALIZATION_TYPE_TABLE,
        appUrl: id => `dhis-web-pivot/?id=${id}`,
        appName: i18n.t('Pivot Tables'),
    },
    [CHART]: {
        id: CHART,
        endPointName: 'charts',
        propName: 'chart',
        pluralTitle: i18n.t('Charts'),
        plugin: 'chartPlugin',
        domainType: DOMAIN_TYPE_AGGREGATE,
        visualizationType: VISUALIZATION_TYPE_CHART,
        appUrl: id => `dhis-web-data-visualizer/#/${id}`,
        appName: i18n.t('Visualizer'),
    },
    [MAP]: {
        id: MAP,
        endPointName: 'maps',
        propName: 'map',
        pluralTitle: i18n.t('Maps'),
        plugin: 'mapPlugin',
        domainType: DOMAIN_TYPE_AGGREGATE,
        visualizationType: VISUALIZATION_TYPE_MAP,
        appUrl: id => `dhis-web-maps/?id=${id}`,
        appName: i18n.t('Maps'),
    },
    [EVENT_REPORT]: {
        id: EVENT_REPORT,
        endPointName: 'eventReports',
        propName: 'eventReport',
        pluralTitle: i18n.t('Event reports'),
        plugin: 'eventReportPlugin',
        domainType: DOMAIN_TYPE_TRACKER,
        visualizationType: VISUALIZATION_TYPE_TABLE,
        appUrl: id => `dhis-web-event-reports/?id=${id}`,
        appName: i18n.t('Event Reports'),
    },
    [EVENT_CHART]: {
        id: EVENT_CHART,
        endPointName: 'eventCharts',
        propName: 'eventChart',
        pluralTitle: i18n.t('Event charts'),
        plugin: 'eventChartPlugin',
        domainType: DOMAIN_TYPE_TRACKER,
        visualizationType: VISUALIZATION_TYPE_CHART,
        appUrl: id => `dhis-web-event-visualizer/?id=${id}`,
        appName: i18n.t('Event Visualizer'),
    },
    [APP]: {
        endPointName: 'apps',
        propName: 'appKey',
        pluralTitle: i18n.t('Apps'),
    },
    [REPORTS]: {
        id: REPORTS,
        endPointName: 'reports',
        propName: 'reports',
        pluralTitle: i18n.t('Reports'),
        appUrl: id =>
            `dhis-web-reporting/getReportParams.action?mode=report&uid=${id}`,
    },
    [RESOURCES]: {
        id: RESOURCES,
        endPointName: 'resources',
        propName: 'resources',
        pluralTitle: i18n.t('Resources'),
        appUrl: id => `api/documents/${id}/data`,
    },
    [USERS]: {
        id: USERS,
        endPointName: 'users',
        propName: 'users',
        pluralTitle: i18n.t('Users'),
        appUrl: id => `dhis-web-dashboard-integration/profile.action?id=${id}`,
    },
    [TEXT]: {
        id: TEXT,
        propName: 'text',
    },
    [MESSAGES]: {
        propName: 'messages',
    },
    [SPACER]: {
        propName: 'text',
    },
};

export const getEndPointName = type => itemTypeMap[type].endPointName;

export const getItemUrl = (type, item, d2) => {
    let url;

    if (type === APP) {
        url = item.launchUrl;
    }

    if (itemTypeMap[type] && itemTypeMap[type].appUrl) {
        url = `${getBaseUrl(d2)}/${itemTypeMap[type].appUrl(item.id)}`;
    }

    return url;
};

export const getItemIcon = type => {
    switch (type) {
        case REPORT_TABLE:
        case EVENT_REPORT:
        case REPORTS:
            return TableIcon;
        case CHART:
        case EVENT_CHART:
            return ChartIcon;
        case MAP:
            return MapIcon;
        case APP:
            return ExtensionIcon;
        case RESOURCES:
            return DescriptionIcon;
        case USERS:
            return PersonIcon;
        case TEXT:
            return FontDownloadIcon;
        case MESSAGES:
            return EmailIcon;
        case SPACER:
            return CropFreeIcon;
        default:
            return NotInterestedIcon;
    }
};
