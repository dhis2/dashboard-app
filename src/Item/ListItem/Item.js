import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { List, ListItem as MUIListItem } from 'material-ui/List';
import IconButton from 'material-ui/IconButton';
import SvgIcon from 'd2-ui/lib/svg-icon/SvgIcon';
import ItemHeader from '../ItemHeader';
import Line from '../../widgets/Line';
import { REPORTS, RESOURCES, USERS, itemTypeMap } from '../../itemTypes';
import { orArray } from '../../util';
import { tRemoveListItemContent } from './actions';
import { colors } from '../../colors';

const getItemTitle = item => {
    return itemTypeMap[item.type].pluralTitle;
};

const getContentItems = item =>
    orArray(item[itemTypeMap[item.type].propName]).filter(
        (item, index, array) =>
            array.findIndex(el => el.id === item.id) === index
    );

const getLink = (item, id, context) => {
    let url;

    switch (item.type) {
        case REPORTS:
            url = `dhis-web-reporting/getReportParams.action?mode=report&uid=${id}`;
            break;
        case RESOURCES:
            url = `api/documents/${id}/data`;
            break;
        case USERS:
            url = `dhis-web-messaging/profile.action?uid=${id}`;
            break;
        default:
            break;
    }

    return `${context.baseUrl}/${url}`;
};

const removeContent = (handler, item, contentToRemove) => () => {
    handler(item, contentToRemove);
};

const ListItem = (props, context) => {
    const { item, editMode, tRemoveListItemContent } = props;

    // avoid showing duplicates
    const contentItems = getContentItems(item);

    const primaryText = contentItem => {
        const deleteButton = (
            <IconButton
                style={{
                    verticalAlign: 'text-bottom',
                    height: '32px',
                }}
                iconStyle={{
                    width: 20,
                    height: 20,
                    fill: colors.red,
                }}
                onClick={removeContent(
                    tRemoveListItemContent,
                    item,
                    contentItem
                )}
            >
                <SvgIcon icon="Delete" />
            </IconButton>
        );

        return (
            <div>
                <a
                    style={{ textDecoration: 'none' }}
                    href={getLink(item, contentItem.id, context)}
                >
                    {contentItem.name}
                </a>
                {editMode ? deleteButton : null}
            </div>
        );
    };

    return (
        <Fragment>
            <ItemHeader title={getItemTitle(item)} />
            <Line />
            <List className="dashboard-item-content">
                {contentItems.map(contentItem => (
                    <MUIListItem
                        key={contentItem.id}
                        primaryText={primaryText(contentItem)}
                    />
                ))}
            </List>
        </Fragment>
    );
};

ListItem.contextTypes = {
    baseUrl: PropTypes.string,
};

const ListItemContainer = connect(null, {
    tRemoveListItemContent,
})(ListItem);

export default ListItemContainer;
