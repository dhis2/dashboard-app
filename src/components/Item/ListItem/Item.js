import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { colors } from '@dhis2/ui-core'
import DescriptionIcon from '../../../icons/Description'

import DeleteIcon from '../../../icons/Delete'
import Line from '../../../widgets/Line'
import { itemTypeMap, getItemUrl } from '../../../modules/itemTypes'
import { orArray } from '../../../modules/util'
import { tRemoveListItemContent } from './actions'
import ItemHeader from '../ItemHeader'
import classes from './Item.module.css'

const getItemTitle = item => itemTypeMap[item.type].pluralTitle

const getContentItems = item =>
    orArray(item[itemTypeMap[item.type].propName]).filter(
        (item, index, array) =>
            array.findIndex(el => el.id === item.id) === index
    )

const ListItem = (props, context) => {
    const { item, editMode, tRemoveListItemContent } = props
    const contentItems = getContentItems(item)

    const getLink = contentItem => {
        const deleteButton = (
            <button
                className={classes.deletebutton}
                onClick={() => tRemoveListItemContent(item, contentItem)}
            >
                <DeleteIcon className={classes.deleteicon} />
            </button>
        )

        return (
            <>
                <a
                    className={classes.link}
                    style={{ color: colors.grey900 }}
                    href={getItemUrl(item.type, contentItem, context.d2)}
                >
                    {contentItem.name}
                </a>
                {editMode ? deleteButton : null}
            </>
        )
    }

    return (
        <>
            <ItemHeader title={getItemTitle(item)} itemId={item.id} />
            <Line />
            <div className="dashboard-item-content">
                <ul className={classes.list}>
                    {contentItems.map(contentItem => (
                        <li className={classes.item} key={contentItem.id}>
                            <DescriptionIcon className={classes.itemicon} />
                            {getLink(contentItem)}
                        </li>
                    ))}
                </ul>
            </div>
        </>
    )
}

ListItem.propTypes = {
    editMode: PropTypes.bool,
    item: PropTypes.object,
    tRemoveListItemContent: PropTypes.func,
}

ListItem.contextTypes = {
    d2: PropTypes.object,
}

export default connect(null, {
    tRemoveListItemContent,
})(ListItem)
