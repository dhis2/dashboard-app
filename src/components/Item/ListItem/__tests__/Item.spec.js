import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { RESOURCES } from '../../../../modules/itemTypes.js'
import ListItem from '../Item.js'

const mockStore = configureMockStore([])

describe('ListItem component', () => {
    let store

    beforeEach(() => {
        store = mockStore({})
    })

    it('renders correctly in view mode with content items', () => {
        const item = {
            id: 'item1',
            type: RESOURCES,
            shortened: false,
            resources: [
                { id: 'resource1', name: 'Resource 1' },
                { id: 'resource2', name: 'Resource 2' },
            ],
        }

        const { asFragment } = render(
            <Provider store={store}>
                <ListItem
                    item={item}
                    dashboardMode="view"
                    isFullscreen={false}
                    isSlideshowView={false}
                />
            </Provider>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders correctly in fullscreen mode', () => {
        const item = {
            id: 'item3',
            type: RESOURCES,
            shortened: false,
            resources: [
                { id: 'resource1', name: 'Resource 1' },
                { id: 'resource2', name: 'Resource 2' },
            ],
        }

        const { asFragment } = render(
            <Provider store={store}>
                <ListItem
                    item={item}
                    dashboardMode="view"
                    isFullscreen={true}
                    isSlideshowView={false}
                />
            </Provider>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders correctly with no content items', () => {
        const item = {
            id: 'item4',
            type: RESOURCES,
            shortened: false,
            resources: [],
        }

        const { asFragment } = render(
            <Provider store={store}>
                <ListItem
                    item={item}
                    dashboardMode="view"
                    isFullscreen={false}
                    isSlideshowView={false}
                />
            </Provider>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
