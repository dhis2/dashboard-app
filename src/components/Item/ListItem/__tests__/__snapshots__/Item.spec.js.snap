// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders correctly in fullscreen mode 1`] = `
<DocumentFragment>
  <div
    class="itemHeaderWrap"
  >
    <p
      class="itemTitle"
    >
      Resources
    </p>
    <div
      class="itemHeaderRightWrap"
    />
  </div>
  <div
    class="jsx-742385706 jsx-3230761442 "
    data-test="dhis2-uicore-divider"
    role="separator"
  />
  <div
    class="content fullscreen"
  >
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <span
          class="itemContent"
        >
          <svg
            color="#6C7787"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 1l4 4v9a1 1 0 01-1 1H4a1 1 0 01-1-1V2a1 1 0 011-1zM8 2H4v12h8V6H8zm-.5 9a.5.5 0 110 1h-2a.5.5 0 110-1zm1-2a.5.5 0 010 1h-3a.5.5 0 010-1zm2-2a.5.5 0 110 1h-5a.5.5 0 010-1zM9 2.415V5h2.586z"
              fill="currentColor"
            />
          </svg>
          <a
            class="link"
            href="../api/documents/resource1/data"
            rel="noreferrer"
            style="color: rgb(33, 41, 52);"
            tabindex="0"
            target="_top"
          >
            Resource 1
          </a>
        </span>
      </li>
      <li
        class="item"
      >
        <span
          class="itemContent"
        >
          <svg
            color="#6C7787"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 1l4 4v9a1 1 0 01-1 1H4a1 1 0 01-1-1V2a1 1 0 011-1zM8 2H4v12h8V6H8zm-.5 9a.5.5 0 110 1h-2a.5.5 0 110-1zm1-2a.5.5 0 010 1h-3a.5.5 0 010-1zm2-2a.5.5 0 110 1h-5a.5.5 0 010-1zM9 2.415V5h2.586z"
              fill="currentColor"
            />
          </svg>
          <a
            class="link"
            href="../api/documents/resource2/data"
            rel="noreferrer"
            style="color: rgb(33, 41, 52);"
            tabindex="0"
            target="_top"
          >
            Resource 2
          </a>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`ListItem component renders correctly in view mode with content items 1`] = `
<DocumentFragment>
  <div
    class="itemHeaderWrap"
  >
    <p
      class="itemTitle"
    >
      Resources
    </p>
    <div
      class="itemHeaderRightWrap"
    />
  </div>
  <div
    class="jsx-742385706 jsx-3230761442 "
    data-test="dhis2-uicore-divider"
    role="separator"
  />
  <div
    class="content"
  >
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <span
          class="itemContent"
        >
          <svg
            color="#6C7787"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 1l4 4v9a1 1 0 01-1 1H4a1 1 0 01-1-1V2a1 1 0 011-1zM8 2H4v12h8V6H8zm-.5 9a.5.5 0 110 1h-2a.5.5 0 110-1zm1-2a.5.5 0 010 1h-3a.5.5 0 010-1zm2-2a.5.5 0 110 1h-5a.5.5 0 010-1zM9 2.415V5h2.586z"
              fill="currentColor"
            />
          </svg>
          <a
            class="link"
            href="../api/documents/resource1/data"
            rel="noreferrer"
            style="color: rgb(33, 41, 52);"
            tabindex="0"
            target="_top"
          >
            Resource 1
          </a>
        </span>
      </li>
      <li
        class="item"
      >
        <span
          class="itemContent"
        >
          <svg
            color="#6C7787"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 1l4 4v9a1 1 0 01-1 1H4a1 1 0 01-1-1V2a1 1 0 011-1zM8 2H4v12h8V6H8zm-.5 9a.5.5 0 110 1h-2a.5.5 0 110-1zm1-2a.5.5 0 010 1h-3a.5.5 0 010-1zm2-2a.5.5 0 110 1h-5a.5.5 0 010-1zM9 2.415V5h2.586z"
              fill="currentColor"
            />
          </svg>
          <a
            class="link"
            href="../api/documents/resource2/data"
            rel="noreferrer"
            style="color: rgb(33, 41, 52);"
            tabindex="0"
            target="_top"
          >
            Resource 2
          </a>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`ListItem component renders correctly with no content items 1`] = `
<DocumentFragment>
  <div
    class="itemHeaderWrap"
  >
    <p
      class="itemTitle"
    >
      Resources
    </p>
    <div
      class="itemHeaderRightWrap"
    />
  </div>
  <div
    class="jsx-742385706 jsx-3230761442 "
    data-test="dhis2-uicore-divider"
    role="separator"
  />
  <div
    class="content"
  >
    <ul
      class="list"
    />
  </div>
</DocumentFragment>
`;
